keras_cv/utils/preprocessing.py

```diff
@@ -323,3 +324,51 @@ def check_fill_mode_and_interpolation(fill_mode, interpolation):
             "Unknown `interpolation` {}. Only `nearest` and "
             "`bilinear` are supported.".format(interpolation)
         )
+
+
+def clip_bounding_box(bounding_boxes, bounding_box_format, images):
+    """clips bounding boxes to image boundaries"""
+    boxes_rank = len(bounding_boxes.shape)
+    images_rank = len(images.shape)
+    boxes_includes_batch = boxes_rank == 3
+    images_include_batch = images_rank == 4
+    if boxes_includes_batch != images_include_batch:
+        raise ValueError(
+            "clip_bounding_box() expects both boxes and images to be batched, or both "
+            f"boxes and images to be unbatched.  Received len(boxes.shape)={boxes_rank}, "
+            f"len(images.shape)={images_rank}.  Expected either len(boxes.shape)=2 AND "
+            "len(images.shape)=3, or len(boxes.shape)=3 AND len(images.shape)=4."
+        )
+
+    def clip_unbatched_bbox(bbox):
```

Comment:
you're dropping the additional axes here after y2.  please include `rest` as a final assignment from the split operation.

Please see https://github.com/keras-team/keras-cv/blob/master/keras_cv/bounding_box/converters.py#L26 as an example.

Reply:
Done!